indico/modules/rb_new/client/js/common/map/MapMarkers.jsx

```diff
@@ -0,0 +1,115 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {connect} from 'react-redux';
+import {Marker, Tooltip} from 'react-leaflet';
+import Leaflet from 'leaflet';
+import MarkerClusterGroup from 'react-leaflet-markercluster';
+import * as mapSelectors from './selectors';
+
+
+function groupIconCreateFunction(cluster) {
+    const highlight = cluster.getAllChildMarkers().some(m => m.options.highlight);
+    return Leaflet.divIcon({
+        html: `<span>${cluster.getChildCount()}</span>`,
+        className: `marker-cluster marker-cluster-small rb-map-cluster ${highlight ? 'highlight' : ''}`,
+        iconSize: Leaflet.point(40, 40, true),
+    });
+}
+
+const icon = Leaflet.divIcon({className: 'rb-map-marker', iconSize: [20, 20]});
+const hoveredIcon = Leaflet.divIcon({className: 'rb-map-marker highlight', iconSize: [20, 20]});
+
+/**
+ * This wrapper is used to avoid re-rendering of <Marker> on every parent
+ * cycle. By adding MarkerWrapper, we gain control of that through
+ * shouldComponentUpdate(...)
+ */
+class MarkerWrapper extends React.Component {
```

Comment:
Don't you need to declare `highlight` as `propTypes` here?

Reply:
Indeed. I wonder why ESLint didn't complain...

Reply:
Oh, it did (Travis), it's my local ESLint that is silly.